test/e2e/user_management_test.go

```diff
@@ -42,7 +45,99 @@ func (s *userManagementTestSuite) TearDownTest() {
 	s.testCtx.Cleanup()
 }
 
+func (s *userManagementTestSuite) TestUserDeactivation() {
+	s.setApprovalPolicyConfig("automatic")
+
+	userSignup, mur := s.createAndCheckUserSignup(true, "iris-at-redhat-com", "iris@redhat.com", approvedByAdmin()...)
+
+	// Deactivate the user
+	userSignup.Spec.Deactivated = true
+	err := s.awaitility.Client.Update(context.TODO(), userSignup)
+	require.NoError(s.T(), err)
+	s.T().Logf("user signup '%s' set to deactivated", userSignup.Name)
+
+	err = s.hostAwait.WaitUntilMasterUserRecordDeleted(mur.Name)
+	require.NoError(s.T(), err)
+}
```

Comment:
Could you please also test a case when the user is reactivated?

Reply:
Sure thing, I've added a test for this.

Reply:
thanks, but it would be great if you could add it as a subtest of the `ban provisioned usersignup` test - it would be much easier to read

Reply:
Hm.. why not as a subtest for the deactivation test as is?

Reply:
because this is easier to read and undrestand what is the target of the tests:
```go
s.T().Run("ban provisioned usersignup", func(t *testing.T) {
	s.setApprovalPolicyConfig("automatic")

	// Create a new UserSignup and confirm it was approved automatically
	userSignup, mur := s.createUserSignupAndAssertAutoApproval(false)

	// Create the BannedUser
	bannedUser := s.createAndCheckBannedUser(userSignup.Annotations[v1alpha1.UserSignupUserEmailAnnotationKey])

	// Confirm the user is banned
	_, err := s.hostAwait.WithRetryOptions(wait.TimeoutOption(time.Second*10)).WaitForUserSignup(userSignup.Name,
		wait.UntilUserSignupHasConditions(approvedAutomaticallyAndBanned()...))
	require.NoError(s.T(), err)

	// Confirm that a MasterUserRecord is deleted
	_, err = s.hostAwait.WithRetryOptions(wait.TimeoutOption(time.Second * 10)).WaitForMasterUserRecord(userSignup.Spec.Username)
	require.Error(s.T(), err)

	t.Run("unban the banned user", func(t *testing.T) {
		// Unban the user
		err = s.awaitility.Client.Delete(context.TODO(), bannedUser)
		require.NoError(s.T(), err)

		// Confirm the BannedUser is deleted
		err = s.hostAwait.WaitUntilBannedUserDeleted(bannedUser.Name)
		require.NoError(s.T(), err)

		// Confirm the user is provisioned
		_, err = s.hostAwait.WithRetryOptions(wait.TimeoutOption(time.Second*10)).WaitForUserSignup(userSignup.Name,
			wait.UntilUserSignupHasConditions(approvedAutomatically()...))
		require.NoError(s.T(), err)

		// Confirm the MUR is created
		_, err = s.hostAwait.WaitForMasterUserRecord(mur.Name)
		require.NoError(s.T(), err)
	})
})
```
just my point of view ;-)

Reply:
OK. You are refereeing to different tests :)
You comment was added to (and mentioned) the deactivation/re-activation tests. But you actually meant ban/unban tests. Now it makes sense.
Here we go: https://github.com/codeready-toolchain/toolchain-e2e/pull/80/commits/61dcbceaf564b1e3b70190332c9f052e94a77aaa

Reply:
wrong copy-paste with the name of the test :man_shrugging: sorry for that
and thanks for changing it :-)